code/modules/mob/living/silicon/robot/robot.dm

```diff
@@ -757,6 +757,7 @@ GLOBAL_LIST_INIT(robot_verbs_default, list(
 			//This will mean that removing and replacing a power cell will repair the mount, but I don't care at this point. ~Z
 			C.brute_damage = 0
 			C.electronics_damage = 0
+			module.update_cells(W)
```

Comment:
While youre doing this, have it null out the baton call when its taken out of the borg itself. 

Failure to do so is going to hold a ref to the cell, which if this cell is destroyed before a new one is put in, thats going to cause excess GC load.